### PR TITLE
Default item selection and callback for lists that get populated dynamically

### PR DESCRIPTION
**Pull request by Noor0**
I came across a problem where I needed to show items in the dropdown that were being fetched from the server and had to set first item as selected by default. The selection was used to fetch more items so i added a callback too that get called whenever the list is populated for the first time.

### CODE REVIEW

**Comment by lfkwtz** [comment]
i don't understand the purpose of this callback. your parent component can tell you when the async action is completed, at which point the picker items will update if you set them to state and read off that state.

if you want to set an item to be selected, set the `value` prop to be that item.

**Comment by Noor0** [comment]
yeah you're but instead of moving that logic up in the component hierarchy i though it would be better to encapsulate it inside the component.

**Comment by lfkwtz** [comment]
Got it. Appreciate the PR - but I want to avoid opening up this rabbit hole where we have an overly complex API for every possible callback / request, when this is easily solved in the parent.

Also, in the future if you open a PR, please ensure you're on the most recent version. This PR was hard to parse since there were so many changes as you're on an old version of the library.
